### ISSUE TITLE
support routing policy setting for ARecord class

### ISSUE THREAD

**Issue by geekybaiyi** [aws/aws-cdk#4391]
Please support routing policy parameters for ARecord in CDK.

Not all parameters are supported in CDK Route53 module currently. https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-route53-recordset.html

any plan to support those in the future?

---

This is a :rocket: Feature Request

**Comment by NGL321** [comment]
Hi @geekybaiyi,

Thank you for submitting the request! From what I could tell in the [docs](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-route53.RecordSetProps.html), it looks like all the parameters supported in Cloudformation are supported in the CDK.

Could you tell me which parameters you are missing?

**Comment by geekybaiyi** [comment]
how about these ?
"Failover" , "GeoLocation", "MultiValueAnswer", "Region", "SetIdentifier", "Weight"

**Comment by nmussy** [comment]
Indeed, they are missing from the L2 `ResourdSet` construct. You can access them right now via the [`CfnRecordSet`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-route53.CfnRecordSet.html) resource, for example:

```ts
const zone = HostedZone.fromLookup(this, 'zone', {domainName: 'example.com'});

const record = new ARecord(this, 'A', {
    zone,
    target: RecordTarget.fromIpAddresses('1.1.1.1'),
    recordName: 'www',
});

const recordSet = (record.node.defaultChild as CfnRecordSet);
recordSet.failover = 'PRIMARY';
```

**Comment by geekybaiyi** [comment]
I noticed an PR is created associated with this issue. 
any idea when it would be merged?  thx!

**Comment by itaykat** [comment]
How can one reach CfnRecordSet from an ApplicationLoadBalancedFargateService object?

**Comment by ryanvaloris** [comment]
The [CfnRecordSet](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-route53.CfnRecordSet.html) seems to be missing the Latency routing policy as well.  Is there a timeline for when that will be added?

**Comment by jdnz** [comment]
@ryanvaloris you can enable latency routing by setting the `region` and `setIdentifier` of the [CfnRecordSet](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-route53.CfnRecordSet.html), e.g.
```
const record = new ARecord(this, 'apiA', {
      zone: zone,
      target: target,
      recordName: apiName,
});

const recordSet = (record.node.defaultChild as CfnRecordSet);
recordSet.region = region;
recordSet.setIdentifier = region;
```

**Comment by theGreatHeisenberg** [comment]
Is there a way to create cross region Arecord or CNAME? I want to implement latency based routing where I will creating CNAMEs for latency based routing.  

```
parentZone CNAME IADZone
parentZone CNAME PDXZone
```

Where `parentZone`, `IADZone`, and `PDXZone` are all in different AWS accounts . Is there a way to create this via CDK? Possibly using `CrossAccountZoneDelegationRecord`?

**Comment by eugene-screenmeet** [comment]
> @ryanvaloris you can enable latency routing by setting the `region` and `setIdentifier` of the [CfnRecordSet](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-route53.CfnRecordSet.html), e.g.
> 
> ```js
> const record = new ARecord(this, 'apiA', {
>     zone: zone,
>     target: target,
>     recordName: apiName,
> });
> 
> const recordSet = (record.node.defaultChild as CfnRecordSet);
> recordSet.region = region;
> recordSet.setIdentifier = region;
> ```

You the real MVP

**Comment by shadogar** [comment]
We shouldn't have to be going into the L1 construct like this. The L2 should support all of the routing policies. Is this ticket still on the roadmap? I see it's been around for 3 years now

**Comment by andremmfaria** [comment]
Bumping this feature request. This L2 construct would be awesome to have.

**Comment by UriConMur** [comment]
Hi, Im expecting to be able to setup the type as weighted and Im doing it this way:

```javascript
const aRecord = new ARecord(this.scope, resourceId, {
      zone: this.hostedZone,
      target: RecordTarget.fromAlias(new CloudFrontTarget(distribution)),
      recordName: domain,
    })
    const recordSet = aRecord.node.defaultChild as CfnRecordSet
    recordSet.setIdentifier = 'Weighted' // this needs to be unique from all your other A Weighted Records
    recordSet.weight = 200
```

And it works fine, it will be easier of we can add this using constructor props.

**Comment by renatoargh** [comment]
It's already more than 2 years since [jdnz](https://github.com/jdnz) amazing answer... do we know if there's already a nicer way to support latency-based A records from CDK? 👀

**Comment by badmintoncryer** [comment]
Hi there.
I've implemented weighted routing (#28705) and latency based routing (#28723). Additionally, I plan to implement features such as IP-based routing. 
I believe it will still take some time before the merge, but please wait a little longer.

**Comment by badmintoncryer** [comment]
Hi there.
I've implemented weighted routing (https://github.com/aws/aws-cdk/pull/28705), so you can configure it easily from L2 now.

Additionally, I plan to implement features such as latency based routing (https://github.com/aws/aws-cdk/pull/28723) and IP-based routing.

I believe it will still take some time before the merge, but please wait a little longer.

**Comment by flaxpanda** [comment]
> Hi there. I've implemented weighted routing (#28705) for RecordSet class, so you can configure it easily from L2 now. It is possible to set not only A records but all other types of records as well.
> 
> Additionally, I plan to implement features such as latency based routing (#28723) and IP-based routing(#28833) or so.
> 
> I believe it will still take some time before the merge, but please wait a little longer.

thanks @badmintoncryer!

**Comment by wladyslawczyzewski** [comment]
hi there. I have already implemented the Failover routing strategy - but haven't published it yet as it has some dependencies to health checks on which I'm working on rn #9481
